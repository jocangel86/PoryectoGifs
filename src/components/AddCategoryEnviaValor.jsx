import { useState } from "react"
import PropTypes from 'prop-types';


// En esta implementación, se recibe como prop la fución que agrega la categoria onAddCategorie
// Y se le manda el valor de la caja para que la reciba en la función y haga el setInputValue
export const AddCategories=({onAddCategorie, setCategories})=>{

    const [inputValue, setInputValue] = useState('')

    const onChangeValue = (e) =>{
        setInputValue(e.target.value)
    }

    const enviar = (e)=>{
        e.preventDefault();

        if(inputValue.trim().length <= 1) return;

        console.log(inputValue)
        // setCategories(categories => [...categories, inputValue]);



        onAddCategorie(inputValue)
        setInputValue('')
    }

    return(
        <>
        <form onSubmit={enviar} aria-label="form">
            <input  type="text" 
                    placeholder="Buscar gifs / Agregar categoría" 
                    value = {inputValue}
                    onChange={onChangeValue}
                    />
            {/* <button  onClick={onAddCategorie} >Add</button> */}
        </ form>
        <button  onClick={()=> setCategories([])} >Limpiar búsquedas</button>
        </>

    )
}

AddCategories.propTypes={
    onAddCategorie: PropTypes.func.isRequired
}