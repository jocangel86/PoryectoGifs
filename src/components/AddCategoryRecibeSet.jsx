import { useState } from "react"

// En esta implementación, se recibe como prop la fución setInputValue y se afecta directamente el estado.
export const AddCategories=({setCategories})=>{

    const [inputValue, setInputValue] = useState('')

    const onChangeValue = (e) =>{
        setInputValue(e.target.value)
    }

    const enviar = (e)=>{
        e.preventDefault();

        if(inputValue.trim().length <= 1) return;

        console.log(inputValue)
        setCategories(categories => [...categories, inputValue]);
        setInputValue('')
    }

    return(
        <form onSubmit={enviar}>
            <p>Add Categorie:</p>
            <input  type="text" 
                    placeholder="Buscar gifs" 
                    value = {inputValue}
                    onChange={onChangeValue}
                    />
            {/* <button  onClick={(event)=>onAddCategorie(event)} >Add</button> */}
            {/* <button  onClick={onAddCategorie} >Add</button> */}
        </ form>
    )
}