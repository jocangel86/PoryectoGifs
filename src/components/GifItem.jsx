import PropTypes from 'prop-types';

export const GifItem=({title, url})=>{
    // Puedes recibir {img} o desestructurar el prop {title, url}
    return(
        <div  className="card">
              {/* <p>id: {gif.id}</p> */}
              <p>{title}</p>
              <img src={url} alt={title}/>
            </div>
    )
}

GifItem.propTypes={
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
}