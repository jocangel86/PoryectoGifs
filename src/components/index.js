
export * from './AddCategoryEnviaValor';
// export * from './AddCategoryRecibeSet';
export * from './GifGrid';
export * from './GifItem';