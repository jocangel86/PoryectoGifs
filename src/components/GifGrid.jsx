import { GifItem } from "./GifItem";
import { useFetchGif } from "../hooks/useFetchGif";
import { useState } from "react";
import PropTypes from 'prop-types';

export const GifGrid=({ category, categories, setCategories })=> {
  // Se generó el custoHook

  const { gifs, isLoading, offset, setOffset } = useFetchGif(category);
  //const {offset, setOffset} = useState(category.offset)

  //console.log({ gifs, isLoading, offset });

  const eliminaCategoria = (nombre)=>{
   

    //Filtrar para eliminar del array la que recibo
    let newCategories = categories.filter(cat => cat !== nombre ) 
    //Actualizar estado listado
    setCategories(newCategories)

  }



  return (
    <>
      <h3>{category}</h3>
      {/* <p>Los gifs de ésta categoría:</p> */}

{isLoading &&<div>Cargando...</div> }

      <div className="card-grid">
        {gifs.map((gif) => {
          return (
            <GifItem key={gif.id} {...gif}/>
            //  {...gif} es igual al  de abajo, manda todas las props de gif.
            // El anterior se usa cuando tienes muchas propiedades por mandar.
            //  gif={gif}
          );
        })}

        {offset>0 &&     
        <button onClick={() => { setOffset(offset - 10); }}>-- Anterior </button>
        }
        |
        <button onClick={() => {eliminaCategoria(category);}}>-- Eliminar -- </button>
        |
        <button
          onClick={() => {
            setOffset(offset + 10);
          }}
        >
          Siguiente --
        </button>
      </div>
    </>
  );
}

GifGrid.propTypes={
  category: PropTypes.string.isRequired,
}