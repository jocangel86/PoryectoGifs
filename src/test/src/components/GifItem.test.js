import {render, screen} from '@testing-library/react'
import { GifItem } from '../../../components'

describe('Pruebas en GiftItem',()=>{

    const title='Hola mundo';
    const url='https://github.com/'

    test('Debe hacer match con el Snapshot PropTypes',()=>{
        
        const{container} = render(<GifItem title={title} url={url}/>)

        expect(container).toMatchSnapshot();     
    })

    test('Debe mostrar la imagen con el url indicado',()=>{
        
        const{container} = render(<GifItem title={title} url={url}/>)
        //screen.debug();
        const {alt, url} = screen.getByRole('img')
        expect(url).toBe(url);    
        expect(alt).toBe(title);     
    })

    test('Debe mostrar el tÃ­tulo el compoente',()=>{    
        render(<GifItem title={title} url={url}/>)
        expect(screen.getByText(title)).toBeTruthy();      
    })
    
})