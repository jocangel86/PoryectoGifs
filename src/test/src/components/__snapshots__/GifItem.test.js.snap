// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GiftItem Debe hacer match con el Snapshot PropTypes 1`] = `
<div>
  <div
    class="card"
  >
    <p>
      Hola mundo
    </p>
    <img
      alt="Hola mundo"
      src="https://github.com/"
    />
  </div>
</div>
`;
